src/backend/distributed/utils/resource_lock.c

```diff
@@ -1104,22 +1113,14 @@ AcquireDistributedLockOnRelations(List *relationIdList, LOCKMODE lockMode)
 	 */
 	bool startedLockCommand = false;
 
+	/* Used to terminate partial lock commands */
+	const char *lockTerminationString = nowait ? " IN %s MODE NOWAIT;\n" :
```

Comment:
aren't we sure that no code-path can ever pass `nowait` to `lock_relation_if_exists`? I mean,
- We only use LOCK_RELATION_IF_EXISTS for foreign tables
- Foreign TABLEs do NOT support LOCK command:
```SQL
BEGIN;
LOCK foreign_table IN SHARE MODE;
ERROR:  "foreign_table" is not a table or view
Time: 0.292 ms
```

So, my suggestion would be to throw an error if `nowait` is true?

With that, we can avoid the changes on the LOCK_RELATION_IF_EXISTS, and avoid confusion. Also, when that is supported, we can remove the error check and implement what you implement now. But, seems not necessary at this point

Reply:
(same for ONLY and * for now)

Reply:
There is no code path that triggers `lock_relation_if_exists(..., nowait => true)` unless the user explicitly calls that.
There are a few things that can be done about this:
1. Remove the nowait support from `lock_relation_if_exists` since we are not using it anyway.
2. Throw an error inside `lock_relation_if_exists` if `nowait` is true. I think this is counterintuitive to the udf since what is the point of having a boolean parameter if one of its values is disallowed.
3. Throw an error inside `AcquireDistributedLockOnRelations_Internal` when a foreign table is being locked with `nowait` true. Not sure what is appropriate for the error to say.

Let me know what you think is best.

Reply:
I think we can do (1) and (3) together. The error could be `this feature is currently not implemented` or such. As only Citus devs would be able to trigger the error when they refactor things to hit the error, I wouldn't worry about much